### ISSUE TITLE
Write definition of Singleton pattern in README.md

### ISSUE THREAD

**Issue by ash-teach** [nic-dgl104-winter-2024/pattern-library#20]
- [ ] Identify one or more appropriate references.
- [ ] Write one or two paragraphs describing the **purpose** of Singleton pattern (no code necessary).
- [ ] Include well-formatted references (either immediately following the description, or in a references section).

**Comment by kattsky** [comment]
Hi @ash-teach Kindly assign this task to me. Thank you

**Comment by ash-teach** [comment]
Done. FYI: @ting-dev-coder did some work on this in [patterns/singleton](https://github.com/nic-dgl104-winter-2024/pattern-library/blob/main/patterns/singleton/README.md). 

I'm on the fence whether or not we should have individual README.md docs in each pattern directory, as demonstrated in the link above. On the one hand, having documentation in too many places may result in maintenance challenges. On the other hand, separating out the documentation in this way allows us to be more detailed in each individual folder (with summary information in our main [README.md file](https://github.com/nic-dgl104-winter-2024/pattern-library/blob/main/README.md)).

Opinions welcome from anyone, but also those who have weighed in on this topic already - @ting-dev-coder and @KaiKitKobold? Please comment here if you have any thoughts.

Either way, @kattsky, I think we need at least some summary info in the main README.md file, so feel free to write your own, or excerpt from what is already in the pattern/singleton folder.

**Comment by ting-dev-coder** [comment]
Maybe we could still maintain one README.md for each pattern and use comments with a certain format to provide a summary?  The format is open to discussion, by the way. Here's just a sample:
```
/**
 * Singleton example, implemented in JavaScript.
 * @param {number} a - Description of param1
 * @param {number} b - Description of param2
 * @returns {number} - Description of return value
 */
```

**Comment by kattsky** [comment]
Thank you @ash-teach, I will add a summary in main README.md

**Comment by kattsky** [comment]
Update: Singleton definition ready to upload to main README

**Comment by KaiKitKobold** [comment]
> Done. FYI: @ting-dev-coder did some work on this in [patterns/singleton](https://github.com/nic-dgl104-winter-2024/pattern-library/blob/main/patterns/singleton/README.md).
> 
> I'm on the fence whether or not we should have individual README.md docs in each pattern directory, as demonstrated in the link above. On the one hand, having documentation in too many places may result in maintenance challenges. On the other hand, separating out the documentation in this way allows us to be more detailed in each individual folder (with summary information in our main [README.md file](https://github.com/nic-dgl104-winter-2024/pattern-library/blob/main/README.md)).
> 
> Opinions welcome from anyone, but also those who have weighed in on this topic already - @ting-dev-coder and @KaiKitKobold? Please comment here if you have any thoughts.
> 
> Either way, @kattsky, I think we need at least some summary info in the main README.md file, so feel free to write your own, or excerpt from what is already in the pattern/singleton folder.

Oh, I only included the Read.me's because Github has a weird thing, that if an folder is empty, it doesn't get committed into the project, It's essentially only there so that the folders appear until everyone does their work, then it's fairly easy to delete them.

**Comment by kattsky** [comment]
Hi @KaiKitKobold Thank you for your reply. I've seen your work in Singleton README, and it was awesome. As for the main README it now includes Documentation Table of Contents.  I've noticed that the Observer Pattern Definition has been completed, and I'm interested in contributing by adding a few lines about the Singleton pattern.

**Comment by ash-teach** [comment]
Please do go ahead with this @kattsky. I've merged a few PRs recently so the main README.md has changed somewhat, so you might want to sync up to the latest changes. Or you can just submit your work and I'll merge the conflicts.

**Comment by ash-teach** [comment]
Closed by #70.
